libs/content/workspace/src/workspace.rs

```diff
@@ -53,6 +56,10 @@ pub struct Workspace {
 
     pub status: WsStatus,
     pub out: Response,
+
+    pub knowledge_graph: Option<KnowledgeGraphApp>,
+    pub data_given: bool,
+    pub run_graph: bool,
```

Comment:
I think all three of these are best modeled on the Tab rather than on Workspace. What if I want to open two knowledge graphs at the same time, each in a different folder? Lmk if you need help with this data model change